pensieve/pre_treatment.py

```diff
@@ -0,0 +1,26 @@
+import attr
+import cattr
+from pandas import DataFrame
+
+
+@attr.s(auto_attribs=True)
+class PreTreatment:
+    """Represents an abstract pre-treatment step applied to data before calculating statistics."""
+
+    @classmethod
+    def name(cls):
+        return __name__  # todo: snake case names?
+
+    def apply(self, df: DataFrame):
```

Comment:
I think the column name probably needs to be included -- unless you're thinking we'd subset the data frame down to the rows we cared about first.